extensions/applicationinsights-clickanalytics-js/src/events/PageAction.ts

```diff
@@ -21,23 +21,26 @@ export class PageAction extends WebEvent {
         ext['web'] = {};
         let event: ITelemetryItem = {
             name: "Microsoft.ApplicationInsights.{0}.Event",
-            baseType: 'ClickEvent',
+            baseType: 'EventData',
             ext,
             data: {},
             baseData: {}
         };
 
         this._populateEventDataIfPresent(event.baseData, 'name', pageActionEvent.name);
-        this._populateEventDataIfPresent(event.baseData, 'uri', pageActionEvent.uri);
-        this._populateEventDataIfPresent(event.baseData, 'pageType', pageActionEvent.pageType);
-        this._populateEventDataIfPresent(event.baseData, 'properties', pageActionEvent.properties);
-        this._populateEventDataIfPresent(event.baseData, 'actionType', pageActionEvent.actionType);
-        this._populateEventDataIfPresent(event.baseData, 'behavior', pageActionEvent.behavior);
-        this._populateEventDataIfPresent(event.baseData, 'clickCoordinates', pageActionEvent.clickCoordinates);
-        this._populateEventDataIfPresent(event.baseData, 'content', pageActionEvent.content);
-        this._populateEventDataIfPresent(event.baseData, 'targetUri', pageActionEvent.targetUri);
+        this._populateEventDataIfPresent(event.data, 'baseTypeSource', 'ClickEvent');
+        this._populateEventDataIfPresent(event.data, 'uri', pageActionEvent.uri);
+        this._populateEventDataIfPresent(event.data, 'pageType', pageActionEvent.pageType);
+        this._populateEventDataIfPresent(event.data, 'properties', pageActionEvent.properties);
+        this._populateEventDataIfPresent(event.data, 'actionType', pageActionEvent.actionType);
+        this._populateEventDataIfPresent(event.data, 'behavior', pageActionEvent.behavior);
+        this._populateEventDataIfPresent(event.data, 'clickCoordinates', pageActionEvent.clickCoordinates);
+        this._populateEventDataIfPresent(event.data, 'content', pageActionEvent.content);
+        this._populateEventDataIfPresent(event.data, 'targetUri', pageActionEvent.targetUri);
         this._populateEventDataIfPresent(event.data, 'timeToAction', pageActionEvent.timeToAction);
         this._populateEventDataIfPresent(event.data, 'refUri', pageActionEvent.refUri);
+        this._populateEventDataIfPresent(event.data, 'pageName', pageActionEvent.pageName);
+        this._populateEventDataIfPresent(event.data, 'parentId', pageActionEvent.parentId);
         for (let property in properties) {
```

Comment:
As properties is optional it might not exist.
These 2 lines (for..in and hasOwnProperty) can be replased by using objForEachKey() (from CoreUtils) which also checks if the target (properties) exists.